src/pages/all-rules.js

```diff
@@ -0,0 +1,271 @@
+          className="w-full text-left"
+        >
+          {title}{' '}
+          <span className="number">
+            <p
+              style={{
+                textTransform: 'none',
+                display: 'inline-block',
+              }}
+            ></p>
+          </span>
+          <span className="collapse-icon">
+            <FontAwesomeIcon
+              icon={isAscending ? faSortAmountUp : faSortAmountDown}
+            />
+          </span>
+        </button>
+      </h6>
+      <ol className={'pt-3 px-4 py-2 block'}>{children}</ol>
+    </>
+  );
+};
+
+const AllRulesContent = ({
+  filteredItems,
+  title,
+  notFound,
+  isAscending,
+  setIsAscending,
+}) => {
+  const formatDistanceLocale = {
+    lessThanXSeconds: '{{count}} sec',
+    xSeconds: '{{count}} sec',
+    halfAMinute: '30s',
+    lessThanXMinutes: '{{count}} min',
+    xMinutes: '{{count}} min',
+    aboutXHours: '{{count}} hour',
+    xHours: '{{count}} hour',
+    xDays: '{{count}} day',
+    aboutXWeeks: '{{count}} week',
+    xWeeks: '{{count}} week',
+    aboutXMonths: '{{count}} month',
+    xMonths: '{{count}} month',
+    aboutXYears: '{{count}} year',
+    xYears: '{{count}} year',
+    overXYears: '{{count}} year',
+    almostXYears: '{{count}} year',
+  };
+
+  const formatDistance = (token, count, options) => {
+    options = options || {};
+
+    const result = formatDistanceLocale[token].replace('{{count}}', count);
+
+    if (options.addSuffix) {
+      if (count > 1) {
+        return result + 's';
+      }
+    }
+
+    return result;
+  };
+
+  return (
+    <section className="mb-5 relative">
+      <Heading
+        title={title}
+        isAscending={isAscending}
+        setIsAscending={setIsAscending}
+      >
+        {!notFound ? (
+          filteredItems.list.map((item, idx) => {
+            return (
+              <div key={idx} className="cat-grid-container">
+                <div className="cat-rule-num">{idx + 1}.</div>
+                <div className="cat-rule-link">
+                  <Link to={`${item.item.fields.slug}`}>
+                    {item.item.frontmatter.title}
+                  </Link>
+                </div>
+                <span className="block all-rules-time">
+                  {filteredItems.filter === FilterOptions.De
+                    ? formatDistanceToNow(
+                        new Date(item.file.node.lastUpdated),
+                        {
+                          locale: {
+                            ...locale,
+                            formatDistance,
+                          },
+                          addSuffix: true,
+                        }
+                      )
+                    : formatDistanceToNow(new Date(item.file.node.created), {
+                        locale: {
+                          ...locale,
+                          formatDistance,
+                        },
+                        addSuffix: true,
```

Comment:
Agreed, It can't however be added into the function there as it's from the date-fns package. I could create a separate component instead if we wanted it to be more readable. what do you think?